arkouda/pdarrayclass.py

```diff
@@ -2622,13 +2622,78 @@ def rotr(x, rot) -> pdarray:
 
 
 @typechecked
-def power(pda: pdarray, pwr: Union[int, float, pdarray]) -> pdarray:
-    return pda**pwr
+def power(pda: pdarray, pwr: Union[int, float, pdarray], where: Union[bool, pdarray] = True) -> pdarray:
+    """
+    Raises an input value or array to a power, can be subjected to a condition that if True will raise value to desiered
+    power and if False will return the original value.
+
+    Notes:
+    This function was modified to more closely match the implementation of numpy's power function. Also fixes
+    an error to make a False condition return the original value, as expected.
 
+    Parameters
+    ----------
+    pda : pdarray
+        A pdarry of values that will be raise to a power (pwr)
+    pwr : integer, float, or pdarray
+        The power(s) that pda is raised to
+    where : Boolean or pdaarray
+        This condition is broadcast over the input. At locations where the condition is True, the corresponding value
+        will be raise to the respective power. Elsewhere, it will retain its original value. Default set to True.
+
+    Returns
+    -------
+        pdarray
+        Returns a pdarray of values raised to a power, under the boolean where condition.
+
+    Examples
+    --------
+    >>> A = ak.arange(5)
+    >>> truth = ak.array([True, False, True, False, True])
+    >>> ak.power(A, 3, truth)
+    array([0, 1, 8, 3, 64])
+    """
+    from arkouda.numeric import where as akwhere
+    from arkouda.numeric import cast as akcast
+    if where is True:
+        return pda**pwr
+    elif where is False:
+        return pda
+    else:
+        exp = pda**pwr
+        return akwhere(where, exp, akcast(pda, dtype(exp)))
 
 @typechecked
-def sqrt(pda: pdarray) -> pdarray:
-    return power(pda, 0.5)
+def sqrt(pda: pdarray, where: Union[bool, pdarray] = True) -> pdarray:
+    """
+    Raises an input value or array to a power, can be subjected to a condition that if True will raise value to desired
+    power and if False will return the original value.
+
+    Notes:
+    This function is just a call to the power function, with a power of .5. See above for more documentation and examples.
+    This function was modified to more closely match the implementation of numpy's power function. Also fixes
+    an error to make a False condition return the original value, as expected.
```

Comment:
Again, just highlighting that this is not actually an error. We should indicate that we deviate from the `NumPy` behavior here instead of stating that we are fixing an error.